app/src/main/java/com/example/spork/login/LoginTabFragment.java

```diff
@@ -0,0 +1,113 @@
+package com.example.spork.login;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+
+import com.example.spork.MainActivity;
+import com.example.spork.R;
+import com.parse.LogInCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+
+public class LoginTabFragment extends Fragment {
+
+    public static final int v = 0;
+    public static final String TAG = "LoginTabFragment";
+
+    private TextView tvUsername;
+    private EditText etUsername;
+    private TextView tvPassword;
+    private EditText etPassword;
+    private TextView tvForgetPassword;
+    private Button btnLogin;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        ViewGroup root = (ViewGroup) inflater.inflate(R.layout.login_tab_fragment, container, false);
+
+        tvUsername = root.findViewById(R.id.tvUsername);
+        etUsername = root.findViewById(R.id.etUsername);
+        tvPassword = root.findViewById(R.id.tvPassword);
+        etPassword = root.findViewById(R.id.etPassword);
+        tvForgetPassword = root.findViewById(R.id.tvForgetPassword);
+        btnLogin = root.findViewById(R.id.btnLogin);
+
+        // animations
+        tvUsername.setTranslationX(800);
+        etUsername.setTranslationX(800);
+        tvPassword.setTranslationX(800);
+        etPassword.setTranslationX(800);
+        tvForgetPassword.setTranslationX(800);
+        btnLogin.setTranslationX(800);
+
+        // opacity
+        tvUsername.setAlpha(v);
+        etUsername.setAlpha(v);
+        tvPassword.setAlpha(v);
+        etPassword.setAlpha(v);
+        tvForgetPassword.setAlpha(v);
+        btnLogin.setAlpha(v);
+
+        tvUsername.animate().translationX(0).alpha(1).setDuration(800).setStartDelay(300).start();
+        etUsername.animate().translationX(0).alpha(1).setDuration(800).setStartDelay(300).start();
+        tvPassword.animate().translationX(0).alpha(1).setDuration(800).setStartDelay(300).start();
+        etPassword.animate().translationX(0).alpha(1).setDuration(800).setStartDelay(300).start();
+        tvForgetPassword.animate().translationX(0).alpha(1).setDuration(800).setStartDelay(300).start();
+        btnLogin.animate().translationX(0).alpha(1).setDuration(800).setStartDelay(300).start();
```

Comment:
Since you are doing the same animation it would be a bit cleaner to try to take the repeated code out to either a separate method or lambda.

In this case, you can simplify since `TextView`/`EditText`/`Button` all extend `View` which the `animate()` method comes from.

The lambda/method could specifically take in a `View` parameter rather than `TextView`/`EditText`/`Button`